tests/test-travis/test-chrome.R

```diff
@@ -32,3 +32,59 @@ assert('chrome_print() works with reveal.js presentations', {
   first_page_text_content = pdftools::pdf_text(f)[1]
   (identical(first_page_text_content, 'Test reveal.js'))
 })
+
+assert('find_gs() finds Ghostscript executable', {
+  (nzchar(find_gs()))
+  (gs_available())
+})
+
+assert('chrome_print() generates expected outline', {
+  f = print_pdf('test-outline.Rmd')
+
+  (is_pdf(f))
+
+  toc = pdftools::pdf_toc(f)[[2L]]
+  res = list(
+    list(title = "1 Title 1", children = list()),
+    list(
+      title = "2 Title 2",
+      children = list(
+        list(title = "2.1 Title 2-1", children = list()),
+        list(title = "2.2 Title 2-2", children = list(
+          list(title = "2.2.1 Title 2-2-1", children = list()),
+          list(title = "2.2.2 Title 2-2-2",children = list())
+        ))
+      )
+    ),
+    list(title = "3 \u4e2d\u6587", children = list()),
+    list(title = "4 \u4e2d\u6587 \u5e26\u7a7a\u683c", children = list(
+      list(title = "4.1 \u4e2d\u6587\u6b21\u7ea7\u6807\u9898", children = list())
+    ))
+  )
+
+  (toc %==% res)
+
+  # works for async
+  # not sure it's the right way to test the promise object
+  f = print_pdf('test-outline.Rmd', async = TRUE)
+  out = promises::then(f, function(f) {
```

Comment:
@RLesur I'm not satisfied with this workaround to test the promises but I can't find a better way. Any better idea?

Reply:
I think the test could wait that the promise is resolved or rejected. For instance, we could use a function like [this one](https://github.com/RLesur/crrri/blob/48b5c2c5e0d209d3742720d30ed91694a6e80c12/R/hold.R#L34-L66).